### PR TITLE
Fix wrong key name in unknown decoration option error

### LINKED ISSUE #1082: `\pgfkeyscurrentname` shown in `unknown key` error messages
**Brief outline of the bug**

With tikz library `decorations` loaded, `\pgfkeyscurrentname`, rather than its value, is shown in error messages if unknown decoration keys are used.

**Minimal working example (MWE)**

```tex
\documentclass{article}
\usepackage{tikz}
\usetikzlibrary{decorations}

\tikzset{decoration={segment length=5pt, red}}
\begin{document}
\end{document}
```

 - Expected: `! Package pgfkeys Error: I do not know the key '/pgf/decoration/red' and ...`
 - Actual: `! Package pgfkeys Error: I do not know the key '/pgf/decoration/\pgfkeyscurrentname ' and ...`

**Analysis**
Caused by
https://github.com/pgf-tikz/pgf/blob/d835e0312028e7e398626bdd77e5effeaab280f0/tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex#L17-L21
and I will provide a PR to fully expand `\pgfkeyscurrentname` before being passed to `/errors/unknown key`.

### PR DESCRIPTION
**Pull request by muzimuzhi**
**Motivation for this change**

Fixes #1082.

Previous PR #1083 tried to fix it but left a wrong key name in the error message, see analysis in https://github.com/pgf-tikz/pgf/pull/1083#discussion_r1178677796 and https://github.com/pgf-tikz/pgf/issues/1082#issuecomment-1524817940.

I also changed it from `pgfkeys` error to `tikz` error.

**Checklist**

Please [signoff your commits][git-s] to explicitly state your agreement to the [Developer Certificate of Origin][DCO]. If that is not possible you may check the boxes below instead:

- [x] Code changes are licensed under [GPLv2][GPL] + [LPPLv1.3c][LPPL]
- [x] Documentation changes are licensed under [FDLv1.2][FDL]

[git-s]: https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s
[DCO]: https://developercertificate.org
[GPL]: https://www.gnu.org/licenses/gpl-2.0.html
[LPPL]: https://www.latex-project.org/lppl/lppl-1-3c.txt
[FDL]: https://www.gnu.org/licenses/fdl-1.2.html

### CODE REVIEW

File: tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex [hunk 1]
```diff
@@ -14,13 +14,15 @@
...
-    {\pgfkeys{/errors/unknown
-        key/.expanded={/pgf/decoration/\pgfkeyscurrentname}{\pgfutil@unexpanded{#1}}}}},%
+  /pgf/decoration/.unknown/.code={%
+    \pgfifdecoration{\pgfkeyscurrentname}
+      {\edef\tikz@decoration@name{\pgfkeyscurrentname}}
+      {\pgfifmetadecoration{\pgfkeyscurrentname}
+        {\edef\tikz@decoration@name{\pgfkeyscurrentname}}
+        {\tikzerror{Unknown decoration option '\pgfkeyscurrentname'.
+            I do not know the key '/pgf/decoration/\pgfkeyscurrentname'
+            and I am going to ignore it. Perhaps you misspelled it}}}%
```
Comment (hmenke) [inline tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex]: Shouldn't we somehow keep the `/errors/unknown key` handler?

Comment (muzimuzhi) [reply tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex]: Using `/error/unknown key` handler, the full error message looks like
```
! Package pgfkeys Error: I do not know the key '/pgf/decoration/red' and 
I am going to ignore it. Perhaps you misspelled it.

See the pgfkeys package documentation for explanation.
```
which seems less helpful as user should really be led to the tikz doc, not pgfkeys doc.

But I confess current `\tikzerror` way is not good too.

Comment (hmenke) [reply tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex]: Then maybe an additional error handler for decorations?

Comment (muzimuzhi) [reply tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex]: I decided to save time and just switch back to `/errors/unknown key`.

**Comment by muzimuzhi** [comment]
Sigh, naming of per-issue tests (after 054f1185) is still not so consistent.

(And sorry for the (force push) noise because I forgot to sign off.)

**Review by hmenke — APPROVED** [review]

**Pull request merged**